integration-libs/opf/cta/core/services/opf-dynamic-cta.service.ts

```diff
@@ -0,0 +1,251 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import {
+  ActiveCartFacade,
+  Cart,
+  CartAddEntrySuccessEvent,
+  CartRemoveEntrySuccessEvent,
+  CartUpdateEntrySuccessEvent,
+} from '@spartacus/cart/base/root';
+import {
+  CurrencyService,
+  EventService,
+  LanguageService,
+  Product,
+  WindowRef,
+} from '@spartacus/core';
+import {
+  CtaEvent,
+  CtaProductItem,
+  CtaScriptsLocation,
+  CtaScriptsRequest,
+  OpfCtaFacade,
+} from '@spartacus/opf/cta/root';
+import { OpfGlobalFunctionsFacade } from '@spartacus/opf/global-functions/root';
+import {
+  GlobalFunctionsDomain,
+  KeyValuePair,
+} from '@spartacus/opf/payment/root';
+import { CurrentProductService } from '@spartacus/storefront';
+import {
+  combineLatest,
+  map,
+  merge,
+  Observable,
+  Subscription,
+  take,
+  tap,
+} from 'rxjs';
+
+@Injectable()
+export class OpfDynamicCtaService {
+  protected globalFunctionsFacade = inject(OpfGlobalFunctionsFacade);
+  protected winRef = inject(WindowRef);
+  protected eventService = inject(EventService);
+  protected currencyService = inject(CurrencyService);
+  protected languageService = inject(LanguageService);
+  protected activeCartFacade = inject(ActiveCartFacade);
+  protected opfCtaFacade = inject(OpfCtaFacade);
+  protected currentProductService = inject(CurrentProductService);
+
+  protected subList: Array<Subscription> = [];
+  protected isOnsiteMessagingInit = false;
+  protected scriptIdentifiers: Array<string> = [];
+  protected isCartPage: boolean;
```

Comment:
[MAJOR]
Do we _really_ have to store this static state property? Now we set it in one methods, which causes side effect on other methods - which is hard to reason about (especially in the async flow of calling those methods).

btw. If we add more CTA flows in the future to this service, then a simple true-or-false variable `isCartPage` might be not sufficient and we might need to come up an enum to describe more possibilities (`cart/product/<some_other>/<some_different_other>`).

If possible, I'd suggest removing this static state completely from this service, and instead pass the `CtaScriptsLocation` as an argument of `initiateEvents()` and `stopEvents()`.